modules/tinymce/src/core/main/ts/newline/InsertAccordion.ts

```diff
@@ -0,0 +1,37 @@
+import Editor from '../api/Editor';
+import * as NewLineUtils from './NewLineUtils';
+
+export const isDetails = (node?: Node): node is HTMLDetailsElement =>
+  node?.nodeName === 'DETAILS';
+
+export const isAccordionBody = (node?: Node | null): node is HTMLDivElement =>
+  node?.nodeName === 'DIV' && (node as HTMLDivElement).classList.contains('mce-accordion-body');
+
+export const getAccordionRoot = (editor: Editor, element: Element): HTMLDetailsElement | null =>
+  editor.dom.getParent(element, isDetails);
+
+export const isInAccordionBody = (editor: Editor, element: Element): boolean =>
+  isAccordionBody(element) || Boolean(editor.dom.getParent(element, isAccordionBody));
+
+export const shouldPreventInsertParagraph = (editor: Editor, shiftKey: boolean, parentBlock: Element): boolean => {
```

Comment:
But this is the behavior intentional for accordion plugin. We've named the plugin like that exactly to distinguish it from the details element. Also won't adding this keyboard behavior for all details elements be a major change then?

Reply:
In theory it could be wider I guess we could discuss this at the stand up. I think it's a bit unlikely that you want to create multiple empty paragraphs at the end of a details element. So if it would work as blockquotes I think that new behavior would be fine.

Reply:
@spocke Updated the file to a more generic `InsertDetails`. It works just for simple cases. However this update made me think — do we really need a `div.mce-accordion-body` within the plugin? What purpose does it serve? Maybe we should stick to the standard and remove the custom accordion body whatsoever?

Reply:
I'm in favour of generic `details` support, but product has been very specific that we want a crafted UI just for accordions inserted by our plugin